src/commonTest/kotlin/org/kbson/corpus/CorpusTests.kt

```diff
@@ -0,0 +1,258 @@
+/*
+ * Copyright 2008-present MongoDB, Inc.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *   http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+@file:Suppress("UNUSED")
+
+package org.kbson.corpus
+
+import com.goncalossilva.resources.Resource
+import kotlin.test.Test
+import kotlin.test.assertEquals
+import kotlin.test.assertFailsWith
+import kotlin.test.fail
+import kotlinx.serialization.SerialName
+import kotlinx.serialization.Serializable
+import kotlinx.serialization.decodeFromString
+import kotlinx.serialization.json.Json
+import org.kbson.BsonDecimal128
+import org.kbson.BsonDocument
+import org.kbson.BsonSerializationException
+import org.kbson.BsonType
+import org.kbson.BsonValue
+import org.kbson.internal.Assertions.assertThrows
+import org.kbson.internal.HexUtils
+
+class ArrayTest : CorpusTest("array.json")
```

Comment:
Why does it have to be this way?  :) I guess there is no way to iterate all the resources in the directory, as in Java?

I couldn't figure out how to run any of the tests in IDEA  just by right clicking on something.

Reply:
I doesn't have to be this way - but there is no resource handling in KMP. So a plugin had to be used and a full path had to be provided to the file.

There is also no Parameterized tests either. You can right click on the classes and get them to run individually or the directory to run all tests in the suite.